### ISSUE TITLE
feature request: possibility of outputting just a few trees

### ISSUE THREAD

**Issue by odanoburu** [GrammaticalFramework/gftest#2]
while writing a new `lin`, it is nice to able to get fast feedback on the results. `gftest` can be somewhat slow depending on the tested function (`:/`), so having a way of getting preliminary results fast would be nice -- specially if easy to implement! if these trees look fine, one can then proceed to doing the full test.

**Comment by inariksit** [comment]
Is this for just the normal mode with `-f SlowFunction`? For -o, I actually have this hack in place https://github.com/GrammaticalFramework/gftest/blob/master/src/Main.hs#L396-L397 , for getting the tested categories to flash by. Checking /tmp/<filename> also works to see that it is creating bunch of trees and testing them.

The function `testFun` returns a full string with all results; this is in order to easier decide whether to write to stdout or to a file. If you want, you could change `testFun` into an IO function; instead of this [unlines](https://github.com/GrammaticalFramework/gftest/blob/master/src/Grammar.hs#L979), make it into `mapM_ print`.

Another thing, if you're not doing it already, definitely compile your PGF with the GF flag `--optimize-pgf`! Basque doesn't even run on my computer without that.

**Comment by odanoburu** [comment]
> Is this for just the normal mode with -f SlowFunction? For -o, I actually have this hack in place 

yes, but I could use `-o` too! are you using strictness to make it run faster?

> Another thing, if you're not doing it already, definitely compile your PGF with the GF flag --optimize-pgf! 

I'm doing it! it should probably be done by default when `--make` is on, given how dramatic a difference it makes!

**Comment by inariksit** [comment]
> are you using strictness to make it run faster?

I have no clue how to use strictness stuff in Haskell :-P that thing about printing to a file was just a random poking around and "huh this seems to work, let's keep it". 

> it should probably be done by default when --make is on, given how dramatic a difference it makes!

Hmm, I was trying to find out if there are reasons why it is not default--here's something in [Krasimir's PhD thesis](http://www.cse.chalmers.se/~krasimir/phd-thesis.pdf) pages 76-77: "The final conclusion is that the optimized and the unoptimized grammars are semantically different." I don't know if that's good enough reason to not have optimized version as default, but at least I learned something new today!
